Paintroid/src/main/java/org/catrobat/paintroid/tools/implementation/BaseToolWithRectangleShape.java

```diff
@@ -364,10 +365,19 @@ protected void drawBitmap(Canvas canvas, float boxWidth, float boxHeight) {
 		canvas.drawBitmap(drawingBitmap, null, tempDrawingRectangle, null);
 	}
 
-	private void drawOverlayBitmap(Canvas canvas, float boxWidth, float boxHeight) {
+	private void drawOverlayBitmap(Canvas canvas, float boxWidth, float boxHeight, float boxRotation) {
 		float size = Math.min(boxWidth, boxHeight) / 8;
 		tempDrawingRectangle.set(-size, -size, size, size);
-		canvas.drawBitmap(overlayBitmap, null, tempDrawingRectangle, null);
+
+		float newWidth = tempDrawingRectangle.width() / overlayBitmap.getWidth();
+		float newHeight = tempDrawingRectangle.height() / overlayBitmap.getHeight();
+
+		Matrix matrix = new Matrix();
```

Comment:
Please do not create the matrix in the loop.
Matrix is just a wrapper for native code, allocating in a loop might be expensive.
Additionally this will trigger the garbage collector more often.
Just reuse the same instance of matrix and call `Matrix.reset()` before using it to clear the last values.